src/sampling.jl

```diff
@@ -206,6 +206,30 @@ end
 self_avoid_sample!(a::AbstractArray, x::AbstractArray) =
     self_avoid_sample!(Base.GLOBAL_RNG, a, x)
 
+"""
+    shrinking_array_sample!([rng], a::AbstractArray, x::AbstractArray)
+
+Shrinking-array sampling: samples from an array by replacing each selected
+element with the last element in the array and decrementing the array length
+counter by 1 each time - hence shrinking-array.
+"""
+function shrinking_array_sample!(rng::AbstractRNG, a::AbstractArray, x::AbstractArray)
+    n = length(a)
+    k = length(x)
+    k <= n || error("length(x) should not exceed length(a)")
```

Comment:
Rather than a generic error this would be a good situation to throw a typed error, e.g. `DimensionMismatch` or `ArgumentError`.